### ISSUE TITLE
Transactions don't seem to be working properly

### ISSUE THREAD

**Issue by denislins** [sdispater/orator#139]
I might be doing something wrong, but currently I'm not able to rollback a transaction. This is the code I'm using:

```python
class TestModels(AppTestCase):
    def test_it_works(self):
        with db.transaction():
            User.create(name='Denis')
            db.rollback()

        assert User.count() == 1
```

The user count continues to grow, however. In addition, I tried raising an exception instead of manually rolling back, with the same result.

Could this be a bug? I am doing something wrong?

**Comment by sdispater** [comment]
Strange. What's your database backend (PostreSQL, MySQL or SQLite)?

**Comment by denislins** [comment]
I'm sorry! I was writing the issue when I got called into a meeting, and I should've been more clear.

I'm using python 3.5.2 and a dockerized postgres v9.6.1, with docker-compose to link the containers. I'm using docker for mac 1.13.0, and OS X 10.12.2.

**Comment by sdispater** [comment]
I just pushed a fix (c4ac034758985ff25146af9637da8b8bd3a2b716) on the `0.9` branch. You can test it to see if it fixes the issue for you.

**Comment by denislins** [comment]
It appears to be working :)

Do you have any idea when will we have this patch released?
